### PR TITLE
adding annotated scatter tab for numeric annotation gene searches (SCP-3209)

### PR DESCRIPTION
**Pull request by devonbush**
Restores annotated scatter plot tab functionality for numeric annotations on gene searches

image

To test:
0: open a study with react_explore on (e.g. the human lymph node synthetic study)
1: Do a gene search
2: switch the annotation to a numeric facet
3: observe that the annotated scatter plot tab appears with the appropriate plot

### CODE REVIEW

**Review by eweitz — APPROVED** [review]

**Review by jlchang — APPROVED** [review]
Thanks for restoring annotated scatter!

File: app/javascript/components/explore/ExploreDisplayTabs.js [hunk 1]
```diff
@@ -441,7 +459,12 @@ export function getEnabledTabs(exploreInfo, exploreParams) {
         enabledTabs = ['dotplot', 'heatmap']
       }
     } else {
-      enabledTabs = ['scatter', 'distribution']
+      if (exploreParams.annotation.type === 'numeric') {
```
Comment (bistline) [inline app/javascript/components/explore/ExploreDisplayTabs.js:462]: I'm going to renew my request that we look at this method and try to figure out a way to simplify the signature.  We're up to 9 different plot/tab types, and this method keeps getting more byzantine as we move forward.

**Review by bistline — APPROVED** [review]
Glad we caught this.  I'm making the case again that we should at least ticket the work to simplify `getEnabledTabs()` - the more tabs that we add to Explore, the more difficult it is to parse the logic of this method.

File: app/javascript/components/explore/ExploreDisplayTabs.js (hunk 1 above)
Comment (devonbush) [reply app/javascript/components/explore/ExploreDisplayTabs.js:462]: agree that we haven't found the optimal form yet.

Comment (devonbush) [reply app/javascript/components/explore/ExploreDisplayTabs.js:462]: agree that we haven't found the optimal form for this yet.

**Pull request merged**